edb/pgsql/parser/ast_builder.py

```diff
@@ -955,8 +955,14 @@ def _build_a_expr(n: Node, c: Context) -> pgast.BaseExpr:
 
 def _build_func_call(n: Node, c: Context) -> pgast.FuncCall:
     n = _unwrap(n, "FuncCall")
+
+    name: Tuple[str, ...] = tuple(_list(n, c, "funcname", _build_str))
+    # Substitute our custom function in place of "format_type".
+    if name == ('pg_catalog', 'format_type'):
+        name = ('edgedb', '_format_type')
```

Comment:
I'm still not sure why this needs to be in the parser. We do exactly the same stuff for `pg_relation_filenode` function in `resolver/static.py`.

And also, if you'd have `SELECT format_type(x)'` that should also resolve to `pg_catalog.format_type`, because pg_catalog is implicitly always the first name in search_path. (we are already handling this in static.py)

Reply:
Now that the rest of it works and is reasonably stable with lots of tests, let me see if I can move it to the resolver where it should logically belong.